### PR TITLE
feat: add ac for taker fee cap in spec 0057

### PR DESCRIPTION
**Pull request by Jiajia-Cui**

### CODE REVIEW

File: protocol/0057-TRAN-transfers.md [hunk 1]
```diff
@@ -388,3 +388,13 @@ If the network parameter `spam.protection.maxUserTransfersPerEpoch` is modified,
...
+
+If the network parameter `cap_reward_fee_multiple` is set, and if some reward is left from the applied cap, the remaining balance should be recalculated only for parties that have not reached their cap until the leftover is less than 1 reward asset unit or the maximum rounds of recalculation is 10
+(<a name="0057-TRAN-072" href="#0057-TRAN-072">0057-TRAN-072</a>)
+
+If the network parameter `cap_reward_fee_multiple` is set, and if all keys are capped, then the remaining balance must be left in the reward pool and included in the distribution in the future epochs. (<a name="0057-TRAN-073" href="#0057-TRAN-073">0057-TRAN-073</a>)
+
+If the scenario above occurs, and the total transferred in a given epoch, this does not affect the size of the next iteration, which proceeds as normal (including decay factors etc.) as if the full transfer has been made (<a name="0057-TRAN-074" href="#0057-TRAN-074">0057-TRAN-074</a>)
```
Comment (Sohill-Patel) [inline protocol/0057-TRAN-transfers.md]: This looks like an additional check to the scenarios above. Either add it to the scenarios it relates to or include the scenario in this ac

File: protocol/0057-TRAN-transfers.md [hunk 2]
```diff
@@ -109,7 +109,7 @@ A party should be able to configure the distribution of rewards by specifying th
...
+- `cap_reward_fee_multiple` [optional] - if set, the actual amount of reward transferred to each public key during distribution for this transfer will be `min(calculated_reward_in_quantum, cap_reward_fee_multiple × fees_paid_this_epoch_in_quantum)`. When calculating how much of the reward each one is getting, if some is left from the applied cap, we recalculate on the remaining balance only for parties that have not reached their cap until the leftover is less than 1 reward asset unit or the maximum rounds of recalculation is 10. If all keys are capped (i.e. the total amount of the transfer cannot be be sent to eligible keys without breaching the cap) then the remaining balance must be left in the reward pool and included in the distribution in future epochs. If this occurs, and the total transferred in a given epoch, this does not affect the size of the next iteration, which proceeds as normal (including decay factors etc.) as if the full transfer has been made.
```
Comment (Sohill-Patel) [inline protocol/0057-TRAN-transfers.md:112]: Should we have valid values ac for this field?

Comment (gordsport) [reply protocol/0057-TRAN-transfers.md:112]: @Sohill-Patel can you propose an AC?

Comment (Sohill-Patel) [reply protocol/0057-TRAN-transfers.md:112]: `cap_reward_fee_multiple` accepts values greater than `"0"`
Do we need/want to have a sensible upper limit

File: protocol/0057-TRAN-transfers.md [hunk 3]
```diff
@@ -388,3 +388,15 @@ If the network parameter `spam.protection.maxUserTransfersPerEpoch` is modified,
...
+
+If the network parameter `cap_reward_fee_multiple` is set, and if some reward is left from the applied cap, the remaining balance should be recalculated only for parties that have not reached their cap until the leftover is less than 1 reward asset unit or the maximum rounds of recalculation is 10
+(<a name="0057-TRAN-072" href="#0057-TRAN-072">0057-TRAN-072</a>)
+
+If the network parameter `cap_reward_fee_multiple` is set, and if all keys are capped, then the remaining balance must be left in the reward pool and included in the distribution in the future epochs. (<a name="0057-TRAN-073" href="#0057-TRAN-073">0057-TRAN-073</a>)
+
+If the network parameter `cap_reward_fee_multiple` is set, and the total reward transferred in a given epoch, this does not affect the size of the next iteration, which proceeds as normal (including decay factors etc.) as if the full transfer has been made (<a name="0057-TRAN-074" href="#0057-TRAN-074">0057-TRAN-074</a>)
+
+If the network parameter `cap_reward_fee_multiple` is set to be `<0`, then it should be rejected (<a name="0057-TRAN-075" href="#0057-TRAN-075">0057-TRAN-075</a>)
```
Comment (Sohill-Patel) [inline protocol/0057-TRAN-transfers.md:402]:
```suggestion
If the network parameter `cap_reward_fee_multiple` is set to be `<=0`, then it should be rejected (<a name="0057-TRAN-075" href="#0057-TRAN-075">0057-TRAN-075</a>)
```

**Review by Sohill-Patel — APPROVED** [review]

**Pull request merged**